### PR TITLE
feat(gradle): Support updating plugin dependencies

### LINKED ISSUE #3534: Gradle plugins not handled
**What Renovate type are you using?**
Renovate self hosted using the official Docker image version 15.10

**Describe the bug**
New versions of gradle plugins are not detected.
The gradle dependencies are working fine though - pull requests for new dependency versions are created as expected.

**Did you see anything helpful in debug logs?**
No

**To Reproduce**
 A repository with a build.gradle containing a plugins section with old plugin versions:
Example:
```groovy
plugins {
    id 'org.springframework.boot' version '2.1.1.RELEASE'
    id 'org.sonarqube' version '2.6.2'
}
```

**Expected behavior**
Expected new pull requests that changes the versions of gradle plugins.

**Screenshots**
N/A

### PR DESCRIPTION
**Pull request by ikesyo**
Closes #3534.

~~I'm waiting that #4148 to be landed.~~

### CODE REVIEW

File: lib/manager/gradle/gradle-updates-report.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ allprojects {
            .collect { "$it.url" }
            .findAll { !it.startsWith('file:') }
         project.repositories = repos
-        def deps = configurations
+        def deps = (buildscript.configurations + configurations)
```
Comment (ikesyo) [inline lib/manager/gradle/gradle-updates-report.js]: > `buildscript.configurations`

This collects the root project's plugins declared by the followings:

```
buildscript {
    ...
    dependencies {
        classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
        classpath 'com.github.jengelman.gradle.plugins:shadow:1.2.3'
        classpath 'com.fkorotkov:gradle-libraries-plugin:0.1'
        classpath "gradle.plugin.se.patrikerdes:gradle-use-latest-versions-plugin:0.2.3"
    }
}
```

and

```
plugins {
    id "com.github.ben-manes.versions" version "0.20.0"
}
```

**Comment by ViceIce** [comment]
Cool, i didn't know, that i can [draft](https://github.blog/2019-02-14-introducing-draft-pull-requests/) a pr

File: lib/manager/gradle/gradle-updates-report.ts [hunk 2]
```diff
@@ -44,7 +44,7 @@ allprojects {
            .collect { "$it.url" }
            .findAll { !it.startsWith('file:') }
         project.repositories = repos
-        def deps = configurations
+        def deps = (buildscript.configurations + configurations)
           .collect { it.dependencies }
```
Comment (ikesyo) [inline lib/manager/gradle/gradle-updates-report.ts]: > `buildscript.configurations`

This collects the root project's plugins declared by the followings:

```
buildscript {
    ...
    dependencies {
        classpath("org.springframework.boot:spring-boot-gradle-plugin:${springBootVersion}")
        classpath 'com.github.jengelman.gradle.plugins:shadow:1.2.3'
        classpath 'com.fkorotkov:gradle-libraries-plugin:0.1'
        classpath "gradle.plugin.se.patrikerdes:gradle-use-latest-versions-plugin:0.2.3"
    }
}
```

and

```
plugins {
    id "com.github.ben-manes.versions" version "0.20.0"
}
```

**Comment by ikesyo** [comment]
@rarkins It's ready for review.

**Pull request merged**